From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Thu, 2 Dec 2021 13:46:26 +0100
Subject: [PATCH] configurable hopper transfer multiplier


diff --git a/src/main/java/net/minecraft/world/level/block/entity/Hopper.java b/src/main/java/net/minecraft/world/level/block/entity/Hopper.java
index 6a1405a8630e90db3b5a3c9152259ba6f5f0c784..2ed79410e04116b153f9c61a7871f18629cd5de9 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/Hopper.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/Hopper.java
@@ -21,4 +21,10 @@ public interface Hopper extends Container {
     double getLevelY();
 
     double getLevelZ();
+
+    // Cheetah start
+    default float getHopperTransferMultiplier() {
+        return 1.0F;
+    }
+    // Cheetah end
 }
diff --git a/src/main/java/net/minecraft/world/level/block/entity/HopperBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/HopperBlockEntity.java
index d241b25acc223982109495542fd17e0fbdf53437..cf8cef05cf9a81bf51e86ef1de4f1d63d6663a79 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/HopperBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/HopperBlockEntity.java
@@ -47,6 +47,7 @@ public class HopperBlockEntity extends RandomizableContainerBlockEntity implemen
     private NonNullList<ItemStack> items;
     private int cooldownTime;
     private long tickedGameTime;
+    private float hopperTransferMultiplier = 1.0F; // Cheetah
 
     // CraftBukkit start - add fields and methods
     public List<HumanEntity> transaction = new java.util.ArrayList<HumanEntity>();
@@ -78,6 +79,16 @@ public class HopperBlockEntity extends RandomizableContainerBlockEntity implemen
     }
     // CraftBukkit end
 
+    // Cheetah start
+    public float getHopperTransferMultiplier() {
+        return hopperTransferMultiplier;
+    }
+
+    public void setHopperTransferMultiplier( float hopperTransferMultiplier ) {
+        this.hopperTransferMultiplier = hopperTransferMultiplier;
+    }
+    // Cheetah end
+
     public HopperBlockEntity(BlockPos pos, BlockState state) {
         super(BlockEntityType.HOPPER, pos, state);
         this.items = NonNullList.withSize(5, ItemStack.EMPTY);
@@ -93,6 +104,11 @@ public class HopperBlockEntity extends RandomizableContainerBlockEntity implemen
         }
 
         this.cooldownTime = nbt.getInt("TransferCooldown");
+        // Cheetah start
+        if(nbt.contains("TransferMultiplier")) {
+            this.hopperTransferMultiplier = nbt.getFloat( "TransferMultiplier");
+        }
+        // Cheetah end
     }
 
     @Override
@@ -103,6 +119,11 @@ public class HopperBlockEntity extends RandomizableContainerBlockEntity implemen
         }
 
         nbt.putInt("TransferCooldown", this.cooldownTime);
+        // Cheetah start
+        if(this.hopperTransferMultiplier != 1.0F) {
+            nbt.putFloat("TransferMultiplier", this.hopperTransferMultiplier);
+        }
+        // Cheetah end
     }
 
     @Override
@@ -205,7 +226,7 @@ public class HopperBlockEntity extends RandomizableContainerBlockEntity implemen
                 ItemStack itemstack = origItemStack;
 
                 final int origCount = origItemStack.getCount();
-                final int moved = Math.min(level.spigotConfig.hopperAmount, origCount);
+                final int moved = Math.min((int)(level.spigotConfig.hopperAmount * hopper.hopperTransferMultiplier), origCount); // Cheetah
                 origItemStack.setCount(moved);
 
                 // We only need to fire the event once to give protection plugins a chance to cancel this event
@@ -241,7 +262,7 @@ public class HopperBlockEntity extends RandomizableContainerBlockEntity implemen
     private static boolean hopperPull(Level level, Hopper ihopper, Container iinventory, ItemStack origItemStack, int i) {
         ItemStack itemstack = origItemStack;
         final int origCount = origItemStack.getCount();
-        final int moved = Math.min(level.spigotConfig.hopperAmount, origCount);
+        final int moved = Math.min((int)(level.spigotConfig.hopperAmount * ihopper.getHopperTransferMultiplier()), origCount); // Cheetah
         itemstack.setCount(moved);
 
         if (!skipPullModeEventFire) {
