From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Sun, 22 May 2022 16:30:58 +0200
Subject: [PATCH] fixed properties loading


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
index ebfaa8d5de5b905bd2dd7778728b8c9939d01252..bad9fe111483c4c64e0847479af77ed1b2c2e328 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
@@ -126,9 +126,9 @@ public class DedicatedServerProperties extends Settings<DedicatedServerPropertie
         }, new JsonObject()), this.get("generate-structures", true), (String) this.get("level-type", (s) -> {
             return s.toLowerCase(Locale.ROOT);
         }, "default"));
-        // Paper start - Configurable rcon ip
-        final String rconIp = this.getStringRaw("rcon.ip");
-        this.rconIp = rconIp == null ? this.serverIp : rconIp;
+        // Paper start - Configurable rcon ip, // Cheetah use get instead of getRaw
+        final String rconIp = this.get("rcon.ip", "");
+        this.rconIp = rconIp.isEmpty() ? this.serverIp : rconIp;
         // Paper end
     }
 
